modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -318,6 +325,31 @@ template<typename _Tp, int n> struct v_reg
     */
     _Tp get0() const { return s[0]; }
 
+    static v_reg<_Tp, n> fromMask(const v_reg<bool, n>& mask)
```

Comment:
Could it be defined as converting constructor? I think that could allow to rebuild existing code without v_mask related updates.
@seiko2plus @alalek WDYT?

Reply:
I don't think so, that will break the type-safety, but I prefer to use converting constructors and operators together for each vector type only so we can avoid the use of sub access to 'val', it silly to keep typing ".val" in every single line. for example, take a look at the defined types in #14261.